### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by jdiouf**
Adds register, login, and logout apis for token authentication

### CODE REVIEW

File: app/core/migrations/0001_initial.py [hunk 1]
```diff
@@ -94,12 +95,34 @@ class Migration(migrations.Migration):
...
             ],
             options={
                 'abstract': False,
             },
         ),
+        migrations.CreateModel(
+            name='XDSUser',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('password', models.CharField(max_length=128, verbose_name='password')),
```
Comment (mattdrayer) [inline app/core/migrations/0001_initial.py:110]: Can we ensure that the passwords are hashed and not stored in clear text?

Comment (rglasgow-deloitte) [reply app/core/migrations/0001_initial.py:110]: Yes, from my understanding. Django Auth handles all that. Ill confirm today and make any changes required.

Comment (jdiouf) [reply app/core/migrations/0001_initial.py:110]: Just validated that the passwords are indeed hashed

Comment (mattdrayer) [reply app/core/migrations/0001_initial.py:110]: Excellent, thanks!

**Comment by jdiouf** [comment]
re-requesting review due to rebase. Everything should still be the same

**Review by KarenAJ — APPROVED** [review]
Approved

**Review by SHGODE — APPROVED** [review]

**Pull request merged**